flink-streaming-java/src/test/java/org/apache/flink/streaming/api/DataStreamTest.java

```diff
@@ -934,6 +936,47 @@ public void onTimer(long timestamp, OnTimerContext ctx, Collector<Integer> out)
         assertThat(getOperatorForDataStream(processed)).isInstanceOf(KeyedProcessOperator.class);
     }
 
+    /**
+     * Verify that a {@link KeyedStream#process(KeyedProcessFunction)} call is correctly translated
+     * to an async operator.
+     */
+    @Test
+    void testAsyncKeyedStreamKeyedProcessTranslation() {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        DataStreamSource<Long> src = env.fromSequence(0, 0);
+
+        KeyedProcessFunction<Long, Long, Integer> keyedProcessFunction =
+                new KeyedProcessFunction<Long, Long, Integer>() {
+                    private static final long serialVersionUID = 1L;
+
+                    @Override
+                    public void processElement(Long value, Context ctx, Collector<Integer> out)
+                            throws Exception {
+                        // Do nothing
```

Comment:
nit: is is worth setting a flag when we come into the processElement, so we can check it has run later?

Reply:
We test whether the transformation is correct in this UT, the checking of `KeyedProcessFunction#processElement` is done in other UTs.